darts/timeseries.py

```diff
@@ -2886,6 +2894,29 @@ def append_values(self, values: np.ndarray) -> Self:
         TimeSeries
             A new TimeSeries with the new values appended
         """
+
+        values = np.array(values) if not isinstance(values, np.ndarray) else values
+
+        # If the component and sample dimension don't match, we try to recast the values into the
+        # appropriate shape:
+        if values.shape[1:] != self._xa.values.shape[1:]:
```

Comment:
As discussed, let's use the existing `expand_arr()`, and then raise an error if the two last dimensions don't match (component, and sample dimension)